### PR TITLE
chore: add auth and model provider configuration documentation

### PR DESCRIPTION
**Pull request by thedadams**

### CODE REVIEW

File: docs/docs/04-configuration/02-model-providers.md [hunk 1]
```diff
@@ -1,2 +1,68 @@
...
+```json
+{
+  "id": "azure-openai-model-provider",
+  "created": "2024-11-08T16:03:21-05:00",
+  "metadata": {
+    "envVars": "OTTO_AZURE_OPENAI_MODEL_PROVIDER_API_KEY,OTTO_AZURE_OPENAI_MODEL_PROVIDER_ENDPOINT,OTTO_AZURE_OPENAI_MODEL_PROVIDER_DEPLOYMENT_NAME"
+  },
+  "name": "Azure OpenAI Provider",
+  "toolType": "modelProvider",
+  "reference": "github.com/otto8-ai/tools/./azure-openai-model-provider",
```
Comment (StrongMonkey) [inline docs/docs/04-configuration/02-model-providers.md]: `github.com/otto8-ai/tools/./azure-openai-model-provider` is this a valid reference that it must have `./` in the middle?

File: docs/docs/04-configuration/02-model-providers.md [hunk 2]
```diff
@@ -1,2 +1,68 @@
 # Model Providers
-Include details about eactly how to configure each supported model provider
\ No newline at end of file
+
+Each package model provider is configured slightly differently, but, at least, the `OTTO_*_MODEL_PROVIDER_API_KEY` environment variable are required, where `*` stands in for the name of the provider (e.g. `OTTO_AZURE_OPENAI_MODEL_PROVIDER_API_KEY`).
```
Comment (drpebcak) [inline docs/docs/04-configuration/02-model-providers.md]:
```suggestion
Each model provider has a slightly different configuration, but all require at least the `OTTO_*_MODEL_PROVIDER_API_KEY` environment variable, where `*` stands in for the name of the provider (e.g. `OTTO_AZURE_OPENAI_MODEL_PROVIDER_API_KEY`).
```

**Review by StrongMonkey — APPROVED** [review]

File: docs/docs/04-configuration/03-auth-providers.md [hunk 3]
```diff
@@ -1,2 +1,30 @@
...
+
+The following environment variables are required for all authentication providers. Setting the Client ID and Client Secret will mean that the authentication provider is enabled. The remaining configuration will be validated on startup.
+
+- `OTTO_AUTH_CLIENT_ID`: The client ID of the authentication provider.
+- `OTTO_AUTH_CLIENT_SECRET`: The client secret of the authentication provider.
+- `OTTO_AUTH_COOKIE_SECRET`: The secret used to encrypt the authentication cookie. Must be of size 16, 24, or 32 bytes.
+- `OTTO_AUTH_ADMIN_EMAILS`: A comma-separated list of the email addresses of the admin users.
+
+The following environment variables are optional for all authentication providers:
+- `OTTO_AUTH_EMAIL_DOMAINS`: The email domains allowed for authentication. Defaults to `*`.
```
Comment (drpebcak) [inline docs/docs/04-configuration/03-auth-providers.md]: I assume this is a comma-separated list just like the admin emails, but we may as well be explicit about it as well

File: docs/docs/04-configuration/03-auth-providers.md [hunk 4]
```diff
@@ -1,2 +1,30 @@
...
+## Google
+
+Google is the default authentication provider. There are currently no additional environment variables required for Google authentication.
+
+## GitHub
+
+GitHub authentication has the following optional configuration:
+
+- `OTTO_AUTH_GITHUB_ORG`: The name of the organization allowed for authentication. Defaults to `*`.
+- `OTTO_AUTH_GITHUB_TEAM`: The name of the team allowed for authentication. Defaults to `*`.
```
Comment (drpebcak) [inline docs/docs/04-configuration/03-auth-providers.md]: Are these also allowed to be comma-separated lists? The wording below for `OTTO_AUTH_GITHUB_ALLOW_USERS` makes it seem like it

**Review by drpebcak — CHANGES_REQUESTED** [review]

File: docs/docs/04-configuration/02-model-providers.md (hunk 1 above)
Comment (thedadams) [reply docs/docs/04-configuration/02-model-providers.md]: This is the way they come out of the API. I change them in the docs for ease of understanding.

File: docs/docs/04-configuration/02-model-providers.md (hunk 2 above)
Comment (thedadams) [reply docs/docs/04-configuration/02-model-providers.md]: Done

File: docs/docs/04-configuration/03-auth-providers.md (hunk 3 above)
Comment (thedadams) [reply docs/docs/04-configuration/03-auth-providers.md]: It is. I changed the wording to make it explicit.

File: docs/docs/04-configuration/03-auth-providers.md (hunk 4 above)
Comment (thedadams) [reply docs/docs/04-configuration/03-auth-providers.md]: No, this can't be a comma-separated list. I changed the wording to make it explicit.

**Review by drpebcak — APPROVED** [review]

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**